lisa/tools/ethtool.py

```diff
@@ -308,65 +324,72 @@ def get_device_link_settings(self, interface: str) -> DeviceLinkSettings:
             return self._device_link_settings_map[interface]
 
         result = self.run(interface)
-        if result.exit_code != 0:
-            raise LisaException(
-                f"ethtool {interface} command got non-zero"
-                f" exit_code: {result.exit_code}"
-                f" stderr: {result.stderr}"
-            )
+        result.assert_exit_code()
 
         device_link_settings = DeviceLinkSettings(interface, result.stdout)
         self._device_link_settings_map[interface] = device_link_settings
 
         return device_link_settings
 
     def get_device_ring_buffer_settings(
-        self, interface: str
+        self, interface: str, force: bool = False
     ) -> DeviceRingBufferSettings:
-        if interface in self._device_ring_buffer_settings_map.keys():
+        if (not force) and (interface in self._device_ring_buffer_settings_map.keys()):
             return self._device_ring_buffer_settings_map[interface]
 
-        result = self.run(f"-g {interface}")
-        if result.exit_code != 0:
-            raise LisaException(
-                f"ethtool -g {interface} command got non-zero"
-                f" exit_code: {result.exit_code}"
-                f" stderr: {result.stderr}"
-            )
+        result = self.run(f"-g {interface}", force_run=force)
+        if (result.exit_code != 0) and ("Operation not supported" in result.stderr):
+            raise SkippedException
+        result.assert_exit_code(
+            message=f"Couldn't get device {interface} ring buffer settings."
+        )
+
+        if interface in self._device_ring_buffer_settings_map.keys():
+            del self._device_ring_buffer_settings_map[interface]
 
         device_ring_buffer_settings = DeviceRingBufferSettings(interface, result.stdout)
         self._device_ring_buffer_settings_map[interface] = device_ring_buffer_settings
 
         return device_ring_buffer_settings
 
+    def change_device_ring_buffer_settings(
+        self, interface: str, rx: int, tx: int
+    ) -> DeviceRingBufferSettings:
+        change_result = self.run(f"-G {interface} rx {rx} tx {tx}")
+        change_result.assert_exit_code(
+            message=f" Couldn't change device {interface} ring buffer settings."
```

Comment:
if the stdout is not long, and can explain what happens, it can be included in output.